code/io/timer.cpp

```diff
@@ -104,7 +106,12 @@ std::uint64_t timer_get_nanoseconds()
 {
 	auto time = get_performance_counter();
 
-	return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
+	if (Timer_perf_counter_freq < NANOSECONDS_PER_SECOND) {
+		// This can cause overflows but I don't know a better solution...
+		return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
+	} else {
+		return time / (Timer_perf_counter_freq / NANOSECONDS_PER_SECOND);
```

Comment:
I don't think that the loss of precision is an issue here since we don't actually need such a high precision timer anywhere. Microseconds are more than enough.

I would have to take a look at the math to determine if GCD would be useful here but if you plan to propose a patch for these things then you could include it there.